# Differential-Gene-Expression-Analysis

# Data

## Microarray data

**Breast cancer**

| GEO datasets | # Breast tumor tissue samples | # Non-tumor breast tissue samples. | Miroarray platform | File type | Description |
| --- | --- | --- | --- | --- | --- |
| GSE29431 | 54 | 12 | GPL570 | Tar (of CEL) |  |
| GSE61304 | 58 | 4 | GPL570 | Tar (of CEL) |  |
| GSE42568 | 104 | 17 | GPL570 | Tar (of CEL) |  |
|  | Total = 216 | Total = 33 |  |  |  |

**Ovarian cancer**
<img width="1000" alt="image" src="https://github.com/Akkun4/Differential-Gene-Expression-Analysis/assets/113637955/7dc7b2c0-52a6-449e-bf81-910dbf190ec6">

Data files were in the .CEL file format. (GSE29431, GSE61304, GSE86374)

> *A CEL file is a data file created by Affymetrix DNA microarray image analysis software. It contains the data extracted from “probes” on an Affymetrix GeneChip and can store thousands of data points, which may make it large in file size.*
> 

GPL570 (HG-U133_Plus_2 - Affymetrix Human Genome U133 Plus 2.0 Array)

## STRING database

200 iterations with 0.5 confidence score


Prob ID to Gene ID/Symbol

<img width="1000" alt="image (1)" src="https://github.com/Akkun4/Differential-Gene-Expression-Analysis/assets/113637955/83b0787a-f342-4674-bda3-8b0d5ac4796b">

- Have to convert to Prob ID to Gene ID/ symbol

| GEO datasets | Annotation |
| --- | --- |
| GSE29431 | hgu133plus2 |
| GSE61304 | hgu133plus2 |
| GSE86374 | pd.hugene.1.0.st.v1 |
- Annotation refers to an information that experiments were conducted on which chip.
    - DNA microarrays are microscope slides that are printed with thousands of tiny spots in defined positions, with each spot containing a known DNA sequence or gene. Often, these slides are referred to as gene chips or DNA chips.
 


<img width="1000" alt="image (2)" src="https://github.com/Akkun4/Differential-Gene-Expression-Analysis/assets/113637955/d409ada3-df0d-42d9-ab6a-7c5e478f7bd9">


Results 
# Data preprocessing

### Breast cancer

Boxplots for the samples in three different datasets → different distributions

![image (4)](https://github.com/Akkun4/Differential-Gene-Expression-Analysis/assets/113637955/33eb4c23-cb26-4787-a4ab-7bd82d33cb3e)

PCA for the samples → three different clusters 

![image (5)](https://github.com/Akkun4/Differential-Gene-Expression-Analysis/assets/113637955/cdf60320-61ba-4ade-9ecf-f0b3ba942b6d)

- technical errors !
- batch effect has affected !

Used Combat in r to handle the batch effect and performed class-specific quantile normalisation (https://www.nature.com/articles/s41598-020-72664-6)

After Batch correction and normalisation

![image (6)](https://github.com/Akkun4/Differential-Gene-Expression-Analysis/assets/113637955/cb45b87d-a24f-4e85-8f9e-0bf53a716e57)


![image (7)](https://github.com/Akkun4/Differential-Gene-Expression-Analysis/assets/113637955/6f2fe6d8-9b84-458b-8d9f-2e47707d1b05)


### Ovarian cancer

Before batch correction and normalization
![image (8)](https://github.com/Akkun4/Differential-Gene-Expression-Analysis/assets/113637955/3923737d-2c18-459f-aa8b-c3d8d130157f)

![image (9)](https://github.com/Akkun4/Differential-Gene-Expression-Analysis/assets/113637955/839fcdbc-15b2-4537-854e-bda667e0b6a0)

After batch correction and normalization

![image (10)](https://github.com/Akkun4/Differential-Gene-Expression-Analysis/assets/113637955/779d9206-15a4-40ea-be32-48f7838ba550)

![image (11)](https://github.com/Akkun4/Differential-Gene-Expression-Analysis/assets/113637955/9fab5cb8-1ad5-486c-890b-828895279a60)


# DGE with limma

- To create gene expression files from .CEL files (for GSE29431, GSE61304, GSE8637), we need ***affy*** and ***limma*** packages from ***bioconductor.***
- **bioconductor *-*** a free, open source and open development software project for the analysis and comprehension of genomic data generated by wet lab experiments in molecular biology.
- **affy -** The package contains functions for exploratory oligonucleotide array analysis.
- **limma -** a library for the analysis of gene expression microarray data, especially the use of linear models for analysing designed experiments and the assessment of differential expression.

Limma Output:
<img width="1000" alt="image (12)" src="https://github.com/Akkun4/Differential-Gene-Expression-Analysis/assets/113637955/bd307cf4-d4af-49aa-a5b7-b0c07002cc88">

- **logFC - Log fold change**
    - Let's say that for gene expression the logFC of B relative to A is 2. If log2(FC) = 2 , the real increase of gene expression from A to B is 4 (2^2) ( FC = 4 ).
    - In other words, A has gene expression four times lower than B, which means at the same time that B has gene expression 4 times higher than A.
- **AveExpr** - the average log2-expression level for that gene across all the arrays and channels in the experiment.
- **t-statistic** - measures the size of the difference relative to the variation in the sample data
- **P value** - a statistical measurement used to validate a hypothesis against observed data.
    - the probability for the experimental outcome as observed or more extreme, if there is no difference in expression between the experimental conditions.
- **adj.P.Val** - The p-value obtained for each gene is re-calculated to correct for running many statistical tests (as many as the number of genes).

- - **B-statistic (lods or B)** - log-odds that that gene is differentially expressed.
    - Suppose for example that B=1.5. The odds of differential expression is exp(1.5)=4.48, i.e, about four and a half to one. The probability that the gene is differentially expressed is 4.48/(1+4.48)=0.82, i.e., the probability is about 82% that this gene is differentially expressed.
    - A B-statistic of zero corresponds to a 50-50 chance that the gene is differentially expressed.
    - The B-statistic is automatically adjusted for multiple testing by assuming that 1% of the genes, or some other percentage specified by the user, are expected to be differentially expressed.
    - If there are no missing values in your data, then the moderated t and B statistics will rank the genes in exactly the same order. Even you do have spot weights or missing data, the p-values and B-statistics will usually provide a very similar ranking of the genes.

- **Additional Information Links:**
1. https://www.rc.fas.harvard.edu/wp-content/uploads/2012/11/Microarray_with_R_and_bioconductor.pdf
2. https://medium.com/@aboutiana/how-to-get-the-differentially-expressed-genes-from-cel-files-in-r-a8a5bbd0aa6f
3. https://www.bioconductor.org/packages/devel/bioc/vignettes/affy/inst/doc/affy.pdf
4. https://kasperdanielhansen.github.io/genbioconductor/html/limma.html


# Breast cancer results

![image (13)](https://github.com/Akkun4/Differential-Gene-Expression-Analysis/assets/113637955/f7ea8660-812a-45c1-a620-fb1927c884a0)

Up → log2FC > 0.5  AND adj.P.Val <0.05

count = 453

Down → log2FC < -0.5  AND adj.P.Val <0.05

count = 130

BRD4 interacted genes 

- Up regulated - `CDK8`
- Down regulated - `RFC2`

Ovarian cancer  results

![image (14)](https://github.com/Akkun4/Differential-Gene-Expression-Analysis/assets/113637955/c0f8b4c9-fc25-42b0-bc15-b171f2451ec1)

Up → log2FC > 0.5  AND adj.P.Val <0.05

count = 449

Down → log2FC < -0.5  AND adj.P.Val <0.05

count = 82

BRD4 interacted genes 

- Up regulated - `{'KDM6B', 'RBM8A', 'RFC2', 'SRCAP'}`
- Down regulated - `{'YY1'}`


